### PR TITLE
Database: Add autoincrement, uniqueness, and sync-writes polyfills

### PR DESCRIPTION
**Pull request by amotl**
## About
Add a few polyfills conceived on behalf of the LangChain adapter, and patches to `rdflib-sqlalchemy` as well as `meltano-target-cratedb`. By adding them here, they can be used by downstream applications without needing to install `cratedb-toolkit`.

## Documentation
Preview: https://sqlalchemy-cratedb--28.org.readthedocs.build/support.html

@simonprickett: I would also be happy about your review on documentation syntax/grammar/wording, as I am not an English native speaker. Thanks!

## References
- https://github.com/crate-workbench/cratedb-toolkit/pull/59
- GH-76
- GH-77
- GH-83

## Backlog
- [x] Software tests.
- [x] Documentation.

### CODE REVIEW

File: src/sqlalchemy_cratedb/polyfill.py
Comment (amotl) [inline src/sqlalchemy_cratedb/polyfill.py:28]: This has become deprecated, and is not needed any longer. There is a better way to write that down.
```python
id = sa.Column(sa.BigInteger, primary_key=True, server_default=sa.func.now())
```

**Review by matriv — COMMENTED** [review]
Thx @amotl !
Nice stuff, left some suggestions on the documentation.

File: docs/support.md
Comment (amotl) [reply docs/support.md:8]: Apparently it's one word. Thanks.

-- https://en.wikipedia.org/wiki/Polyfill_(programming)

File: docs/support.md
Comment (amotl) [reply docs/support.md:65]: Good idea, thanks.

File: docs/support.md
Comment (matriv) [inline docs/support.md:64]: Maybe remove this if you're not planning to fill in this PR.

**Review by matriv — APPROVED** [review]
thx! LGTM, but please wait for one more review.

File: docs/support.md
Comment (amotl) [reply docs/support.md]: I've just updated the patch, with this section completed.

-- https://sqlalchemy-cratedb--28.org.readthedocs.build/support.html#support-for-pandas-and-dask

File: docs/index.rst
Comment (matriv) [inline docs/index.rst:158]: I think that we shouldn't mention as a `distributed OLAP`, specially combining it with `naturally` after that, I would simplify and say that `CrateDB's behavior and features deviate from those found usually in an RDBMS`.

File: docs/overview.rst
Comment (matriv) [inline docs/overview.rst:5]:
```suggestion
Features Overview
```

File: docs/index.rst
Comment (simonprickett) [reply docs/index.rst:158]: Maybe something like "Due to its distributed nature, CrateDB's behavior and features differ from those found in other RDBMS systems".

File: docs/overview.rst
Comment (simonprickett) [inline docs/overview.rst:312]: Not sure what "on behalf of" means here - does this mean "when using the SQLAlchemy dialect" or something else?  "on behalf of" implies there's some ID generating function that can be plugged into SQLAlchemy.  I don't know SQLAlchemy so am not sure what happens here.

File: docs/support.md
Comment (simonprickett) [inline docs/support.md:12]:
```suggestion
you should know what you are doing, as some of them can seriously impact 
```

Removed "the" from "the performance".

File: docs/support.md
Comment (simonprickett) [inline docs/support.md:25]:
```suggestion
INSERT, UPDATE, and DELETE operations for batches of data. It enables
```

Changed to read "It enables bulk operations".

File: docs/support.md
Comment (simonprickett) [inline docs/support.md:161]:
```suggestion
database cluster operations, negating many benefits of using database clusters
```

Minor phrasing update.

**Review by simonprickett — COMMENTED** [review]
Left a couple of small suggestions.

File: docs/overview.rst
Comment (amotl) [reply docs/overview.rst:312]: "when using the SQLAlchemy dialect" is better. Thanks.

File: docs/support.md
Comment (amotl) [reply docs/support.md:25]: Thank you!

File: docs/support.md
Comment (amotl) [reply docs/support.md:161]: Thanks again.

**Comment by amotl** [comment]
@matriv and @simonprickett: Thanks a stack for your excellent reviews.

File: src/sqlalchemy_cratedb/support/polyfill.py
Comment (seut) [inline src/sqlalchemy_cratedb/support/polyfill.py:113]: I think I'd prefer moving this quoting logic into the `refresh_table`function, by change it's signature to consume an optional schema. This would avoid that it will be called with unquoted or invalid relation names.

Comment (amotl) [reply src/sqlalchemy_cratedb/support/polyfill.py:113]: Should work. Thanks!

**Pull request merged**
